llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -1804,6 +1808,11 @@ bool LoopIdiomRecognize::recognizeAndInsertStrLen() {
   // up by later passes
   for (PHINode *PN : Cleanup)
     RecursivelyDeleteDeadPHINode(PN);
+
+  ConstantInt *NewLoopCond = LoopTerm->getSuccessor(0) == LoopBody
```

Comment:
Consider adding a comment explaining that this intends to make the old code unreachable/dead.

Add `assert` that at least one of the successors is `LoopBody`?

Reply:
Thanks for taking a look. I added the asserts and comments.

I will take a look to see if there's any further testing required.